dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/service/ContributionService.java

```diff
@@ -59,31 +56,36 @@ public class ContributionService implements FileService {
     @Value("${services.maat-api.getContributionBatchSize:350}")
     private int getContributionBatchSize;
 
-    @SuppressWarnings("squid:S2147")  // Duplicate code is catch blocks. However they cannot be merged, due to lacking
-    // a shared superclass with .getStatusCode() Same with the throws, to avoid the compiler complaining about throwing
-    // a generic "Exception" back up the chain.
-    public Integer processContributionUpdate(UpdateLogContributionRequest updateLogContributionRequest) {
+    /**
+     * Method which logs that a specific contribution has been processed by the Debt Recovery Company.
+     * <ul>
+     * <li>Will log a success by incrementing the successful count of the associated contribution file.</li>
+     * <li>If error text is present, will instead log it to the MAAT DB as an error for the associated contribution file.</li>
+     * <li>Logs details received in the DCES Event Database.</li>
+     * </ul>
+     * @param contributionProcessedRequest Contains the details of the concor contribution which has been processed by the DRC.
+     * @return FileID of the file associated with the fdcId
+     */
+    public Integer handleContributionProcessedAck(ContributionProcessedRequest contributionProcessedRequest) {
         try {
-            int result;
-            if (!feature.incomingIsolated()) {
-                result = contributionClient.sendLogContributionProcessed(updateLogContributionRequest);
-            } else {
-                log.info("processContributionUpdate: not calling MAAT API sendLogContributionProcessed() because `feature.incoming-isolated=true`");
-                result = 0; // avoid updating MAAT DB.
-            }
-            logContributionAsyncEvent(updateLogContributionRequest, OK);
-            return result;
+            return executeContributionProcessedAckCall(contributionProcessedRequest);
         } catch (WebClientResponseException e){
-            logContributionAsyncEvent(updateLogContributionRequest, e.getStatusCode());
+            logContributionAsyncEvent(contributionProcessedRequest, e.getStatusCode());
             throw e;
         }
     }
 
-    private void logContributionAsyncEvent(UpdateLogContributionRequest updateLogContributionRequest, HttpStatusCode httpStatusCode){
-        BigInteger concorId = BigInteger.valueOf(updateLogContributionRequest.getConcorId());
-        eventService.logConcor(concorId, DRC_ASYNC_RESPONSE, null, null, httpStatusCode, updateLogContributionRequest.getErrorText());
-    }
-
+    /**
+     * Method which will process any Final Defence Cost entries in the correct state for sending to the
```

Comment:
Change "Final Defence Cost" to "Concor Contribution"

Reply:
Gah, good catch. Thanks.